R/LongitudinalGSF.R

```diff
@@ -26,47 +26,94 @@ NULL
+#'
+#' @param psi_bsld (`Prior`)\cr for the baseline value random effect `psi_bsld`. Only used in the
+#'  centered parameterization to set the initial value.
+#' @param psi_ks (`Prior`)\cr for the shrinkage rate random effect `psi_ks`. Only used in the
+#' centered parameterization to set the initial value.
+#' @param psi_kg (`Prior`)\cr for the growth rate random effect `psi_kg`. Only used in the
+#' centered parameterization to set the initial value.
+#' @param psi_phi (`Prior`)\cr for the shrinkage proportion random effect `psi_phi`. Only used in the
+#' centered parameterization to set the initial value.
+#'
+#' @param centered (`logical`)\cr whether to use the centered parameterization.
+#'
 #' @export
 LongitudinalGSF <- function(
-    mu_bsld = prior_lognormal(log(55), 5, init = 55),
-    mu_ks = prior_lognormal(log(0.1), 0.5, init = 0.1),
-    mu_kg = prior_lognormal(log(0.1), 1, init = 0.1),
-    mu_phi = prior_beta(2, 8, init = 0.2),
-    omega_bsld = prior_lognormal(log(0.1), 1, init = 0.1),
-    omega_ks = prior_lognormal(log(0.1), 1, init = 0.1),
-    omega_kg = prior_lognormal(log(0.1), 1, init = 0.1),
-    omega_phi = prior_lognormal(log(0.1), 1, init = 0.1),
-    sigma = prior_lognormal(log(0.1), 0.8, init = 0.1)
+
+    mu_bsld = prior_normal(log(60), 1, init = 60),
+    mu_ks = prior_normal(log(0.5), 1, init = 0.5),
+    mu_kg = prior_normal(log(0.3), 1, init = 0.3),
+
+    omega_bsld = prior_lognormal(log(0.2), 1, init = 0.2),
+    omega_ks = prior_lognormal(log(0.2), 1, init = 0.2),
+    omega_kg = prior_lognormal(log(0.2), 1, init = 0.2),
+
+    a_phi = prior_lognormal(log(5), 1, init = 5),
+    b_phi = prior_lognormal(log(5), 1, init = 5),
+
+    sigma = prior_lognormal(log(0.1), 1, init = 0.1),
+
+    psi_bsld = prior_none(init = 60),
+    psi_ks = prior_none(init = 0.5),
+    psi_kg = prior_none(init = 0.5),
+    psi_phi = prior_none(init = 0.5),
+
+    centered = FALSE
 ) {
-    eta_prior <- prior_std_normal()
+
+    gsf_model <- StanModule(decorated_render(
+        .x = paste0(read_stan("lm-gsf/model.stan"), collapse = "\n"),
+        centered = centered
+    ))
+
+    parameters <- list(
+        Parameter(name = "lm_gsf_mu_bsld", prior = mu_bsld, size = "n_studies"),
+        Parameter(name = "lm_gsf_mu_ks", prior = mu_ks, size = "n_arms"),
+        Parameter(name = "lm_gsf_mu_kg", prior = mu_kg, size = "n_arms"),
+
+        Parameter(name = "lm_gsf_omega_bsld", prior = omega_bsld, size = 1),
+        Parameter(name = "lm_gsf_omega_ks", prior = omega_ks, size = 1),
+        Parameter(name = "lm_gsf_omega_kg", prior = omega_kg, size = 1),
+
+        Parameter(name = "lm_gsf_a_phi", prior = a_phi, size = "n_arms"),
+        Parameter(name = "lm_gsf_b_phi", prior = b_phi, size = "n_arms"),
+        Parameter(name = "lm_gsf_psi_phi", prior = psi_phi, size = "Nind"),
+
+        Parameter(name = "lm_gsf_sigma", prior = sigma, size = 1)
+    )
+
+    if (centered) {
+        parameters_extra <- list(
```

Comment:
better follow pattern `parameters_extra` <- if (...) ... else ...` so move the assignment outside